salt/minion.py

```diff
@@ -3095,6 +3095,9 @@ def _post_master_init(self, master):
         if 'proxy' not in self.opts:
             self.opts['proxy'] = self.opts['pillar']['proxy']
 
+        # update opts, to override data with pillar data
+        self.opts.update(self.opts['pillar'])
```

Comment:
Os should we apply a proper merge and update similar to `config.merge`?

i.e. having

```yaml
my_option:
  my_suboption1: val1
```
in the opts, and:

```yaml
my_option:
  my_suboption2: val2
```

With these changes, will make the opts:

```yaml
my_option:
  my_suboption2: val2
```

Do we want something like below?

```yaml
my_option:
  my_suboption1: val1
  my_suboption2: val2
```